packages/rollup-dev-tools/src/tools/transpiler/binary-search-tree.ts

```diff
@@ -0,0 +1,355 @@
+ *
+ * @param treeNodes The binary search tree representing the nodes to process into bytecode
+ * @param nodeId The nodeId to process all ancestors of.
+ * @param isLeftSibling Whether the given nodeId is a left or right sibling (if a right sibling, it will be JUMPed to.)
+ * @param bytecode The existing EVM bytecode to append to.
+ * @returns The EVM Bytecode with all ancestors of the given nodeId having been added.
+ */
+const appendAncestorsToBytecode = (
+  treeNodes: BinarySearchNode[],
+  nodeId: number,
+  isLeftSibling: boolean,
+  bytecode: EVMBytecode
+): EVMBytecode => {
+  const thisNode: BinarySearchNode = treeNodes[nodeId]
+  log.info(
+    `Processing node with Id ${nodeId} of tree into bytecode, its parameters are ${JSON.stringify(
+      thisNode
+    )}.`
+  )
+  if (isLeafNode(thisNode)) {
+    // If this is a leaf node, we can append the leaf and return since no ancestors.
+    bytecode = [
+      ...bytecode,
+      ...generateLeafBytecode(
+        thisNode as BinarySearchLeafNode,
+        nodeId,
+        isLeftSibling
+      ),
+    ]
+  } else {
+    // Otherwise, we process and append the left and right siblings, recursively using this function.
+    // Append this node
+    const thisInternalNode = thisNode as BinarySearchInternalNode
+    bytecode = [
+      ...bytecode,
+      ...generateInternalNodeBytecode(thisInternalNode, nodeId, isLeftSibling),
+    ]
+    // Append left and right children (left first--right siblings are JUMPed to, left siblings not!)
+    const leftChildId = thisInternalNode.leftChildNodeId
+    const rightChildId = thisInternalNode.rightChildNodeId
+    bytecode = appendAncestorsToBytecode(treeNodes, leftChildId, true, bytecode)
+    bytecode = appendAncestorsToBytecode(
+      treeNodes,
+      rightChildId,
+      false,
+      bytecode
+    )
+  }
+  return bytecode
+}
+
+/**
+ * Fixes all PUSHes tagged by appendAncestorsToBytecode() to correspond to the correct nodes' JUMPDESTs
+ *
+ * @param bytecode The tagged bytecode of the jump search table with incorrect PUSHes for the jumnpdests
+ * @param indexOfThisBlock The offset of this block in the bytecode where the result will be inserted
+ * @param searchTreeNodes All of the binary nodes from which the bytecode was generated
+ * @returns The EVM Bytecode with fixed PUSHes.
+ */
+const fixJUMPsToNodes = (
+  bytecode: EVMBytecode,
+  indexOfThisBlock: number,
+  searchTreeNodes: BinarySearchNode[]
+): EVMBytecode => {
+  for (const opcodeAndBytes of bytecode) {
+    // Find all the PUSHes which we need to append the right sibling's JUMPDEST location to.
+    if (
+      !!opcodeAndBytes.tag &&
+      opcodeAndBytes.tag.reasonTagged === IS_PUSH_BINARY_SEARCH_NODE_LOCATION
+    ) {
+      // Get this node and the node it should JUMP to from the input tree
+      const thisNodeId = opcodeAndBytes.tag.metadata.nodeId
+      const rightChildNodeId = (searchTreeNodes[
```

Comment:
Confused as to why we're only dealing with right children. Also, what if there is no right child?